pkg/endpoint/endpoint.go

```diff
@@ -707,27 +680,48 @@ func (e *Endpoint) GetPolicyModel() *models.EndpointPolicyStatus {
 	e.proxyStatisticsMutex.RUnlock()
 	sortProxyStats(proxyStats)
 
+	var (
+		realizedCIDRPolicy *policy.CIDRPolicy
+		realizedL4Policy   *policy.L4Policy
+	)
+	if e.realizedPolicy != nil {
+		realizedL4Policy = e.realizedPolicy.L4Policy
+		realizedCIDRPolicy = e.realizedPolicy.CIDRPolicy
+	}
+
 	mdl := &models.EndpointPolicy{
 		ID: int64(e.SecurityIdentity.ID),
 		// This field should be removed.
 		Build:                    int64(e.policyRevision),
 		PolicyRevision:           int64(e.policyRevision),
 		AllowedIngressIdentities: realizedIngressIdentities,
 		AllowedEgressIdentities:  realizedEgressIdentities,
-		CidrPolicy:               e.L3Policy.GetModel(),
-		L4:                       e.RealizedL4Policy.GetModel(),
+		CidrPolicy:               realizedCIDRPolicy.GetModel(),
```

Comment:
Oh never mind, I thought this was the desired and the line right after was the realized policy.